tests/test-backend-ops.cpp

```diff
@@ -360,7 +360,10 @@ struct test_case {
         // check if backends support op
         bool supported = true;
         for (ggml_backend_t backend : {backend1, backend2}) {
-            if (!ggml_backend_supports_op(backend, out)) {
+            if (
+                !ggml_backend_supports_op(backend, out)
+                || (op_desc(out) == "MOE" && !strcmp(ggml_backend_name(backend), "Kompute"))
+            ) {
```

Comment:
I would rather have the test fail than add backend-specific exceptions to `test-backend-ops`.

Reply:
Alternatively, all the ops in the graph can be tested with `ggml_backend_supports_op`, instead of only the last one.

Reply:
So when CI is implemented for this backend, it will fail until we implement Mixtral? Or is there some other way to mark the failure as expected?

Reply:
I will change it so that it checks all the ops.

Reply:
I am not convinced about about tests for whole models in `test-backend-ops`. I understand I started this by adding the `test_moe`, initially it was meant as a temporary test to debug an issue while implementing support for mixtral, and maybe it should have been removed after that.

Reply:
I'd be fine with removing them, but it would be great if we had some way to analyze our test coverage. There are several ops/parameters that were being used in the models but not tested in test-backend-ops, and we had no good way to know that until 'main' was failing and 'test-backend-ops' was succeeding.

Reply:
I would like to add graph_compare functionality to `ggml_backend_sched` at some point, and then we could run the same comparison in llama.cpp directly when necessary to debug an issue. Or this could be done in a different test. For now, maybe can do just by not using these tests by default in `test-backend-ops` (just comment the test case in `test_backend`). CUDA is also failing this test currently.

Reply:
I tried to a change to test all the ops for support, but it failed. This is it:
```diff
diff --git a/tests/test-backend-ops.cpp b/tests/test-backend-ops.cpp
index 621e3c45..0dd8e0a0 100644
--- a/tests/test-backend-ops.cpp
+++ b/tests/test-backend-ops.cpp
@@ -357,15 +357,15 @@ struct test_case {
         printf("  %s(%s): ", op_desc(out).c_str(), vars().c_str());
         fflush(stdout);

-        // check if backends support op
+        // check if the backends support the ops
         bool supported = true;
         for (ggml_backend_t backend : {backend1, backend2}) {
-            if (
-                !ggml_backend_supports_op(backend, out)
-                || (op_desc(out) == "MOE" && !strcmp(ggml_backend_name(backend), "Kompute"))
-            ) {
-                printf("not supported [%s] ", ggml_backend_name(backend));
-                supported = false;
+            for (ggml_tensor * t = ggml_get_first_tensor(ctx); t != NULL; t = ggml_get_next_tensor(ctx, t)) {
+                if (!ggml_backend_supports_op(backend, t)) {
+                    printf("not supported [%s] ", ggml_backend_name(backend));
+                    supported = false;
+                    break;
+                }
             }
         }
         if (!supported) {
```

Reply:
Ok, I was able to push it now. For some reason, it doesn't work when using the github CLI to checkout a PR (`gh pr checkout`).

Reply:
Yeah, you can't push to a refs/pull/*/head ref directly (which `gh pr checkout` gives you AFAIK), only to the source branch.